### PR TITLE
Repeat export specifiers on definitions, so that they stay if the declarations are gone

### PR DESCRIPTION
**Pull request by AlexGuteniev**
We should have export specifiers on both function declaration **and** definition.
This is because after some future changes the declaration may be gone - either entirely, or header be no longer included.
By repeating export specifiers we make sure the function type and its impact on export surface stays the same.

See https://github.com/microsoft/STL/pull/3912#issuecomment-1661423190

Additionally, this simplifies the declarations in `stl/src/xmtx.hpp` to match the definitions (the declarations are functionally unchanged).

### CODE REVIEW

**Comment by StephanTLavavej** [comment]
I understand what you're doing here, but can you please provide a PR description so that others can understand (both now, and people in the future doing PR archaeology)? For a PR fixing typos in comments, a title of "Fix typos" with no description would indeed be sufficient, but this PR is doing something much less obvious.

**Comment by AlexGuteniev** [comment]
Updated.

File: stl/src/xmtx.cpp [hunk 1]
```diff
@@ -11,24 +11,24 @@ _EXTERN_C_UNLESS_PURE
 
 // Win32 critical sections are recursive
 
-void __CLRCALL_PURE_OR_CDECL _Mtxinit(_Rmtx* _Mtx) noexcept { // initialize mutex
+_MRTIMP2_PURE void __CLRCALL_PURE_OR_CDECL _Mtxinit(_Rmtx* _Mtx) noexcept { // initialize mutex
```
Comment (StephanTLavavej) [inline stl/src/xmtx.cpp:14]: This is good, but we should simplify `xmtx.hpp` accordingly.

**Review by StephanTLavavej — APPROVED** [review]

File: stl/src/xmtx.cpp (hunk 1 above)
Comment (AlexGuteniev) [reply stl/src/xmtx.cpp:14]: That would be merge conflict with #3900

Comment (StephanTLavavej) [reply stl/src/xmtx.cpp:14]: Trivial to resolve, thanks for the heads up.

**Comment by StephanTLavavej** [comment]
I'm *speculatively* mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Review by CaseyCarter — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I've pushed a merge with `main` to resolve a conflict in `stl/src/xmtx.hpp` where #3900 updated preprocessor comments. Resolved by accepting this PR's elimination of the preprocessor logic.

**Comment by StephanTLavavej** [comment]
Thanks for fixing more of these inconsistencies and defending against future mistakes! :hammer_and_wrench: :shield: :smile_cat:

**Pull request merged**